### DISCUSSION TITLE
Link inbound/outbound direction in case of orthogonal routing

### DISCUSSION THREAD

**Discussion by frnora** [clientIO/joint#2254] [Q&A] [answered]
### Introduction

Hi,

Is there a way to set the links outbound and inbound directions in case of `orthogonal` routing without placing vertices? So something like `startDirections`/`endDirections` in `manhattan`. I would like it to come out always from the `sideNearestToPoint` to the anchor.

Thank you.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
It's not possible with `orthogonal` router. But it is possible with the `rightAngle` router.

_Note: the current version does not support `vertices` but the router from this [PR](https://github.com/clientIO/joint/pull/2224) does._

**Reply by frnora** [reply]
Yep, I tried `rightAngle` but realized vertices were not supported. Looking at the docs now it is actually quite clearly stated there so I don't know whether I am blind or it has just been added recently. I will try the router from the PR, thank you.

**Reply by frnora** [reply]
Just a quick note: definitely an improvement but not quite there yet.

1. Occasionally crosses the element when the vertex is aligned vertically with the anchor.
2. When dragging horizontally, the link creates a loop by the source element that goes away only when an element is set as target. If it is a point, it stays there.

**Reply by kumilingus** [reply]
Would you mind taking a screenshot and pasting it to https://github.com/clientIO/joint/pull/2224? @MartinKanera will look into it.

**Reply by frnora** [reply]
Of course, will do. Give me a while to play around with it and I will report my findings tomorrow. It also seems like the angle of (target/source) markers is wrong. Drawing a straight vertical line (to a point), the marker points to the right. This doesn't happen with the current `rightAngle`.

**Reply by zbynekstara** [reply]
Hi @frnora, additional route improvements to the `rightAngle` router were released as part of JointJS v4.0.2 (JointJS+ v4.0.1) - can you please see if it resolves all issues you experienced?

**Reply by frnora** [reply]
The rotation of the arrowheads seems to be incorrect when the anchor is aligned with the connection points and the distance between the vertex and the target el is not big enough:

https://github.com/clientIO/joint/assets/129097753/46d1189e-5600-4b38-bd85-a3ac6cd1fba2

- notice the arrowhead is actually rendered correctly when the anchor is closer to the bottom (?)
- the radius of the bend around the vertex is also changing when the anchor is dragged up and down

**Reply by MartinKanera** [reply]
Hey @frnora, there is a new PR #2683 introducing several fixes to the `rightAngle` router. Would you mind checking it out to see if the problem persists?
